### PR TITLE
test: Complete test suite for createVenue GraphQL mutation

### PR DESCRIPTION
**Pull request by MonishPuttu**
# test: Complete test suite for createVenue GraphQL mutation

## Description
Added comprehensive integration tests for src/graphql/types/Mutation/createVenue.ts

Fixes: #3864

### Authentication & Authorization (4 tests)
1. Unauthenticated user error
2. Deleted user with valid token (edge case)
3. Non-admin organization member unauthorized
4. Non-member user unauthorized

### Input Validation (2 tests)
5. Invalid UUID/ULID format for organizationId
6. Empty venue name validation

### Resource Validation (1 test)
7. Non-existent organization error

### Business Logic (1 test)
8. Duplicate venue name within organization

### File Upload Validation (2 tests)
9. Invalid MIME type rejection
10. Valid image attachment upload with MinIO storage

### Success Scenarios (7 tests)
11. Platform administrator can create venue
12. Organization administrator member can create venue
13. Venue name reusable across different organizations
14. Nullable fields default to null
15. Special characters and Unicode in venue name
16. Zero capacity value accepted
17. Large capacity value (edge case)

## Test Coverage
- **Statement Coverage:** 95.94%
- **Branch Coverage:** 96.87%
- **Function Coverage:** 100%
- **Uncovered Lines:** 200-209

## Uncovered Lines Explanation

**Lines 200-209** handle a Postgres driver bug where an insert operation succeeds but returns an empty array instead of throwing an error. These lines are intentionally untested because:

1. **Infrastructure-level error:** This handles an external database driver defect, not application logic
2. **Cannot be reproduced:** The code comment states: *"Inserted venue not being returned is an external defect unrelated to this code. It is very unlikely for this error to occur."*
3. **Would require mocking:** Testing this would require mocking the database transaction, which contradicts integration testing principles
4. **Defensive programming:** This is a safety check for an edge case that should never occur in practice

### CODE REVIEW

File: test/graphql/types/documentNodes.ts [hunk 1]
```diff
@@ -2461,3 +2461,22 @@ export const Query_getRecurringEvents = gql(`
...
+            }
+            creator {
+                id
+            }
+            attachments {
+                mimeType
+            }
+        }
+    }
+`);
```
[the remark this answers is not in the document]
Comment (MonishPuttu) [reply test/graphql/types/documentNodes.ts]: working on it

**Comment by MonishPuttu** [comment]
@palisadoes
I've completed 15 integration tests for createVenue achieving 88.46% branch coverage and 100% function coverage, but only 72.07% statement coverage.

Uncovered lines (24-47, 200-209, 212-245) handle:

- Fastify framework-level validation (pre-GraphQL execution)

- Database transaction failures (connection loss, deadlocks)

- MinIO storage service failures during file uploads

These require infrastructure fault injection that's not feasible in integration tests without extensive mocking.

Can I add istanbul ignore next to these lines, or should I document this as intentionally uncovered infrastructure error handling in the PR description?

Thanks!

File: test/graphql/types/Mutation/createVenue.test.ts [hunk 2]
```diff
@@ -0,0 +1,1516 @@
...
+ *
+ * Known Validation Gaps:
+ * - Negative capacity values (e.g., -10) are currently accepted by the API
+ * - Whitespace-only names (e.g., " ") are currently accepted by the API
+ * - Consider adding validation in createVenue.ts resolver for these edge cases
+ *
+ * @remarks
+ * Tests follow talawa-api standards with proper cleanup and isolation.
+ * Each test creates its own test data and cleans up after execution.
+ */
```
[the remark this answers is not in the document]
Comment (MonishPuttu) [reply test/graphql/types/Mutation/createVenue.test.ts:63]: Thanks for the review @coderabbitai

I agree the validation gaps (negative capacity, whitespace-only names) are worth addressing. Since this PR is focused on comprehensive test coverage for the existing implementation, will keep this as a separate issue to track these validation enhancements.

The current tests document these gaps to help future contributors understand what validation is missing, which aligns with the repository's testing standards.

could u perform a comprehensive analysis and let me know if there are any other changes which needs to be made?

**Comment by palisadoes** [comment]
1. Unfortunately, we don't allow istanbul-ignore statements in our code base. It led to gross abuse in the past.
2. You'll need to write the tests, or find another issue to work on and close this PR

**Comment by MonishPuttu** [comment]
ok will try to add those tests
